layouts/events/list.html

```diff
@@ -123,7 +149,118 @@ <h5 class="modal-title" id="suggestionModalLabel">Events & training suggestion</
 </section>
 
 <script>
- // Functions need to be run when the DOM is ready
+  // Initialise session tracking for anti-spam
+  var sessionStartTime = Date.now();
+  var sessionId = generateSessionId('events');
+
+  function submitEventsForm() {
+    // Validate form-specific fields
+    if (!validateEventsForm()) {
+      showEventsError('Please fill out all required fields correctly.');
+      return;
+    }
+    
+    // Validate common submission criteria
+    if (!validateCommonSubmission(sessionStartTime)) {
+      showEventsError('Form submission failed validation. Please try again.');
+      return;
+    }
+    
+    // reCAPTCHA validation - commented out to test anti-spam
+    /*
+    var recaptcha = $("#g-recaptcha-response").val();
+    if (recaptcha === "") {
+      showEventsError("Please tick 'I'm not a robot' above the 'Submit' button.");
+      return;
+    }
+    */
+    
+    // Show loading state
+    setButtonLoadingState('events-submit-btn', true);
+    
+    // Collect form data
+    var formData = new FormData(document.getElementById('events-form'));
+    
+    // Add common metadata
+    addFormMetadata(formData, 'events', sessionId, sessionStartTime);
+    
+    // reCAPTCHA data - commented out to test anti-spam
+    /*
+    var recaptchaResponse = $("#g-recaptcha-response").val();
+    if (recaptchaResponse) {
+      formData.append('g-recaptcha-response', recaptchaResponse);
+    }
+    */
+    
+    // Submit using common function
+    submitFormData(formData, showEventsSuccess, showEventsError)
+    .finally(() => {
+      setButtonLoadingState('events-submit-btn', false);
+    });
+  }
+  
+  // Form-specific validation
+  function validateEventsForm() {
+    var isValid = true;
+    
+    // Check required fields
+    var requiredFields = ['event_title', 'event_type', 'event_start_date', 'event_info_url'];
+    
+    requiredFields.forEach(function(fieldName) {
+      var field = document.getElementById(fieldName);
+      if (!field.value.trim()) {
+        isValid = false;
+        field.classList.add('is-invalid');
+      } else {
+        field.classList.remove('is-invalid');
+      }
+    });
+    
+    // Validate URLs
+    var urlFields = ['event_info_url', 'event_registration_url'];
+    urlFields.forEach(function(fieldName) {
+      var field = document.getElementById(fieldName);
+      if (field.value.trim() && !field.value.match(/^https?:\/\/.+/)) {
+        isValid = false;
+        field.classList.add('is-invalid');
+      } else {
+        field.classList.remove('is-invalid');
+      }
+    });
+    
+    // Validate email
+    var emailField = document.getElementById('submitter_email');
+    if (emailField.value.trim() && !validateEmail(emailField.value)) {
+      isValid = false;
+      emailField.classList.add('is-invalid');
+    } else {
+      emailField.classList.remove('is-invalid');
+    }
+    
+    return isValid;
+  }
+  
+  function showEventsSuccess(message) {
```

Comment:
Can't this be somewhere in common forms js ? if the contact form in a `modal` should behave differently, then if can be achieved by passing a parameter ?